CHANGELOG.md

```diff
@@ -2,9 +2,33 @@
 
 ## v3.11.0-dev
 
+### Enhancements
+
+  * [Ecto.Association] Allow `preload_order` to take MFAs for `many_to_many` associations. This allows ordering by the join table
+  * [Ecto.Query] Add `:operation` option to `with_cte/3`. This allows CTEs to perform updates and deletes
+  * [Ecto.Query] Support `splice(^...)` in `fragment`
+  * [Ecto.Query] Add `:mode` option to `order_by/4`
```

Comment:
I had to consult the docs to remember what `:mode` does. It makes me think it may not be the best option name. Maybe we should instead have `prepend_order_by`? It would be closer to `or_where` and `or_having` in this case. WDYT?

Reply:
agreed! sent a PR